drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/panel/AnalysisReportScreen.java

```diff
@@ -153,4 +164,33 @@ public void showStatus( final Status status ) {
                               status.getEnd(),
                               status.getTotalCheckCount() );
     }
+
+    private Predicate<Issue> shouldNotBeMergedInReport = new Predicate<Issue>() {
+        @Override
+        public boolean test(Issue issue) {
+            return !( issue.getExplanationType() == ExplanationType.EMPTY_RULE ||
+                      issue.getExplanationType() == ExplanationType.MISSING_RESTRICTION ||
+                      issue.getExplanationType() == ExplanationType.MISSING_ACTION );
+        }
+    };
+
+    private void mergeIssues( Set<Issue> targetIssueSet, List<Issue> issues, ExplanationType type) {
```

Comment:
@jhrcek @jomarko  When dealing with the `IssuesSet`, you can merge those two collections by first merging their streams and then applying a collector.

Also, `issuesToMerge`` is needlessly turned into a collection - instead, the `flatMap` on `affectedRows` can be applied directly on the first stream, bypassing the need for the first collection. Although I can see a small benefit in improved readability, IMHO it is so small in this case that it doesn't justify wasting memory and CPU cycles.